docs/architecture/adr-054-crypto-encoding-2.md

```diff
@@ -0,0 +1,83 @@
+# ADR 054: Crypto encoding (part 2)
+
+## Changelog
+
+\*2020-2-27: Created
+
+## Context
+
+Amino has been a pain point of many users in the ecosystem. While Tendermint does not suffer greatly from the performance degradation introduced by amino, we are making an effort in moving the encoding format to a widely adopted format, [Protocol Buffers](https://developers.google.com/protocol-buffers). With this migration a new standard is needed for the encoding of keys. This will cause ecosystem wide breaking changes.
+
+Currently amino encodes keys as `<PrefixBytes> <Length> <ByteArray>`.
+
+## Decision
+
+Transitioning from a fixed size byte array to byte slice's would be the first step. This will enable usage of [cosmos-proto](https://github.com/regen-network/cosmos-proto) interface type. This removes boiler plate needed for oneof types.
```

Comment:
> Transitioning from a fixed size byte array to byte slice's would be the first step.

Can you explain what you mean by this?

NOTE: with bytes array, there's no difference between amino and protobuf (only prefix bytes are different).

amino: (prefix bytes) + varint length + bytes
proto3: (field number + wire_type) + varint length + bytes

https://developers.google.com/protocol-buffers/docs/encoding#strings

Reply:
this is just going from [keySize]byte to []byte.

Reply:
put a small explanation